# ***Setting up the Evnironment***

In [8]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error

# ***Step:1 _ Loading the Dataset***

In [2]:
from google.colab import files
uploaded = files.upload()

Saving boston_cleaned.csv to boston_cleaned.csv


In [3]:
df = pd.read_csv('boston_cleaned.csv')
df.head()

CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD    TAX  \
0  0.00632  18.0   2.31     0  0.538  6.575  65.2  4.0900    1  296.0   
1  0.02731   0.0   7.07     0  0.469  6.421  78.9  4.9671    2  242.0   
2  0.02729   0.0   7.07     0  0.469  7.185  61.1  4.9671    2  242.0   
3  0.03237   0.0   2.18     0  0.458  6.998  45.8  6.0622    3  222.0   
4  0.06905   0.0   2.18     0  0.458  7.147  54.2  6.0622    3  222.0   

   PTRATIO       B  LSTAT  MEDV  
0     15.3  396.90   4.98  24.0  
1     17.8  396.90   9.14  21.6  
2     17.8  392.83   4.03  34.7  
3     18.7  394.63   2.94  33.4  
4     18.7  396.90   5.33  36.2

# ***Step:2 _ Splitting the Dataset***

In [4]:
X = df.drop("MEDV", axis=1)
y = df["MEDV"]

# ***Step:3 _ Performing train/test split***

In [5]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

print("Training set shape:", X_train.shape, y_train.shape)
print("\nTesting set shape:", X_test.shape, y_test.shape)


Training set shape: (214, 13) (214,)

Testing set shape: (54, 13) (54,)


# ***step:4 _ Training a Linear Regression model***

In [7]:
lin_reg = LinearRegression()

lin_reg.fit(X_train, y_train)

LinearRegression()

# ***Step:5 _ Evaluating the Model Performance***

# (a) MAKING PREDICTIONS

In [9]:
y_pred = lin_reg.predict(X_test)

# (b) MEAN SQUARED ERROR

In [20]:
mse = mean_squared_error(y_test, y_pred)
print("MSE : ", mse)

MSE :  3.7281318839818316


`MSE IS (3.72) : It Shows the average squared error. A small value means the model’s predictions are very close to the actual prices.`

# (c) MEAN ABSOLUTE ERROR

In [13]:
mae = mean_absolute_error(y_test, y_pred)
print("MAE : ", mae)

MAE :  1.5151583031541918


`MAE Is (1.51): Its Just On average, predictions are off by about 1.5 house-price units, which is quite low.`

# (d) ROOT MEAN SQUARED ERROR

In [15]:
rmse = np.sqrt(mse)
print("RMSE : ", rmse)

RMSE :  1.9308370941075872


`RMSE is (1.93): The Typical errors are around 2 units, aligning with the MAE and confirming stable performance.`

# (e) MEAN ABSOLUTE PERCENTAGE ERROR

In [17]:
mape = np.mean(np.abs((y_test - y_pred) / y_pred)) * 100
print('MAPE : ', mape)

MAPE :  7.260004705665713


`MAPE is (7.26%): Predictions are off by about 7% on average, which indicates good accuracy since it’s below 10%.`

# **OVERALL PERFORMANCE OF MODEL**
**The Linear Regression model shows strong predictive**
**performance, with low errors across all evaluation metrics.**
**Both absolute and percentage errors (MAE ≈ 1.5, MAPE ≈ 7%)**
**indicate the model predicts house prices with good accuracy.**
**Overall, the model fits the data well and generalizes effectively to unseen test data.**